### ISSUE TITLE
percMethylation does not reach 100%

### ISSUE THREAD

**Issue by SophieValk** [al2na/methylKit#313]
Hi there, 

Thankyou for providing the Methylkit package, I have really enjoyed using it so far. 

I am trying to use MethylKit for identifying differences in DNA Methylation between two groups. After having imported all the sample files independently, I have checked their methylation profiles using the get MethylationStats function. All histograms show varying methylation ranging from 0% - 100%: 
![100pMethylation](https://github.com/al2na/methylKit/assets/106058726/c66aad31-f89e-428f-b4d7-e68120d84968)

$meth <- unite(myobj, destrand=TRUE)
$head(meth)

![headMeth](https://github.com/al2na/methylKit/assets/106058726/f22990c3-03ee-4ee0-8b5e-2e0412116b3a)

When I then try to unite my samples, methylkit returns a methylBase object where the methylation percentage of all samples only reach until 50%. Regardless if I set the destrand option at TRUE or FALSE. 

$PercMethylation <- percMethylation(meth)
$View(PercMethylation)
![Screenshot_percMethylation](https://github.com/al2na/methylKit/assets/106058726/3acab312-27ee-4015-856a-a5a54a603b28)

Am I doing something wrong or am I overseeing something? It seems that somehow in the unite function, the data gets divided by 2. I thought this may have something to do with the destrand option in unite but that did not seem to make a difference. 

When I use destrand=FALSE, I get the following results with the following code:
$meth <- unite(myobj, destrand=FALSE
$PercMethylation <- percMethylation(meth)
$View(PercMethylation)

![percMethylation_destrandFALSE](https://github.com/al2na/methylKit/assets/106058726/8871644d-0154-4e92-a0a9-05cffbc67494)

I hope someone can explain what I might be doing wrong. 
Thanks, 
Sophie

**Comment by alexg9010** [comment]
Hi Sophie,

Thanks for using methylKit!

Unfortunately I was not able to inspect the plots that you uploaded, it
seems that the files are empty.

Best,

Alex


Sophie Valk ***@***.***> schrieb am Mo., 5. Feb. 2024, 08:36:

> Hi there,
>
> Thankyou for providing the Methylkit package, I have really enjoyed using
> it so far.
>
> I am trying to use MethylKit for identifying differences in DNA
> Methylation between two groups. After having imported all the sample files
> independently, I have checked their methylation profiles using the get
> MethylationStats function. All histograms show varying methylation ranging
> from 0% - 100%.
>
> When I then unite the samples per group using the following command:
> methylBase <- unite(myobj, destrand=FALSE)
>
> And then try to plot the methylation percentage using the following:
> percMeth <- percMethylation(methylBase)
> hist(percMeth)
>
> Then the methylation percentage histogram only goes up until 50%. Somehow
> I am not able to import any pictures to this issue, so I provided a
> wetransfer link that shows an example of a methylation percentage histogram
> right after having imported the independent files. And a histogram of the
> percMethylation function after having united my samples per group: https:/
> <https://we.tl/t-GGzcidH7Sy>
>

/we.tl/t-GGzcidH7Sy <https://we.tl/t-GGzcidH7Sy>

**Comment by SophieValk** [comment]
Hi @alexg9010 , 

Thankyou for taking the time to read this issue. I uploaded my photo's in the first comment on this issue. I hope you may be able to tell me what I'm doing wrong. 

Best,
Sophie

**Comment by SophieValk** [comment]
Hi @alexg9010 , 

Thankyou for taking the time to look at my issue. I figured out what was the problem. I did not correctly transform my CpG called .bed files into the required format for methylKit. The freqC and freqT did not add up together. Now that I fixed that the %methylation reaches 100%, also after uniting multiple documents.

Thanks again for providing methylKit, I will happily continue using it. This issue can be closed. 

Best, 
Sophie
